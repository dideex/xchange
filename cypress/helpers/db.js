const mongoose = require('mongoose')
const mongoConfig = require('./config.json').mongodb

module.exports = mongoose.connect(mongoConfig, {useNewUrlParser: true})
