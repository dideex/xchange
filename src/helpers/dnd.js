import React from 'react'
import {DragDropContext} from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'

export function wrapInTestContext(DecoratedComponent) {
  const TestStub = props => <DecoratedComponent {...props} />
  return DragDropContext(TestBackend)(TestStub)
}
