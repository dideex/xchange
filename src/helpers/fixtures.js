export const fakeCurrnecy = [
  {
    order: 1,
    _id: 1,
    icon: 'eth',
    id: 'eth',
    name: 'tether',
    label: 'tether label',
    source: 'tether source wallet'
  },
  {
    order: 2,
    _id: 2,
    icon: 'bitcoin',
    id: 'bitcoin',
    name: 'lisk',
    label: 'lisk label',
    source: 'lisk source wallet'
  },
  {
    order: 3,
    _id: 3,
    icon: 'rur',
    id: 'rur',
    name: 'litecoin',
    label: 'litecoin label',
    source: 'litecoin source wallet'
  },
]


export const fakeData = {
  id: 'Test id',
  inputValue: '1000',
  toWallet: 'Test wallet',
  currencyInputLabel: 'Test input label',
  currencyOutputLabel: 'Test output label',
  fromWallet: 'Tests fromWallet',
  outputValue: '2000',
  email: 'Tests email',
  username: 'Tests username',
  paymentStatus: 1,
  loading: false,
  wallets: {
    eth: 'eth wallet',
    bitcoint: 'bitcoint wallet',
    rur: 'rur wallet',
  },
  login: 'Test login',
  updatePaymentStatus() {},
}

export const fakeUser = {
  username: 'Fake username',
  password: 'Fake password',
  email: 'fake@email.com',
  phone: '89145678901',
  fio: 'Fake fio',
  token: 'Fake token',
}

export const fakeBitcoinAddress = '3JQSigWTCHyBLRD979JWgEtWP5YiiFwcQB '
export const fakeCreditAddress = '1234 4321 5678 8765'

export const fakeWallets = {
  eth: 'Fake eth wallet',
  bitcoin: 'Fake bitcoin wallet',
  rur: 'Fake rur wallet'
}