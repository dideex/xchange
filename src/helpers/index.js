export {fakeCurrnecy} from './fixtures'

export const delay = (fn = () => {}, timeout = 0) =>
  new Promise(res =>
    setTimeout(() => {
      fn()
      res()
    }, timeout),
  )
