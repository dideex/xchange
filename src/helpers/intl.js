import React from 'react'
import {mount, shallow} from 'enzyme'
import {IntlProvider, intlShape} from 'react-intl'

import messages from '../locale/en.json'
const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

function nodeWithIntlProp(node) {
  return React.cloneElement(node, {intl})
}

export function mountWithIntl(
  node,
  {context, childContextTypes, ...additionalOptions} = {},
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, {intl}),
    childContextTypes: Object.assign({}, {intl: intlShape}, childContextTypes),
    ...additionalOptions,
  })
}
