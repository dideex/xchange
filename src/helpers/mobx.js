import React from 'react'
import {Provider} from 'mobx-react'
import {lastOperations, cash, user} from '../store'

jest.mock('../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}}),
  errorEmitter: jest.fn(data => fn => data(fn)),
}))

// jest.mock('../components/common', () => ({
//   noty: () => {},
//   Loading: () => <div>Loading</div>,
// }))

export const MobxProvider = ({children, ...stores}) => {
  return (
    <Provider
      lastOperationsStore={lastOperations}
      cashStore={cash}
      userStore={user}
      {...stores}
    >
      {children}
    </Provider>
  )
}
