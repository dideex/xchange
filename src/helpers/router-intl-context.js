import {mount} from 'enzyme'
import {BrowserRouter} from 'react-router-dom'
import {IntlProvider, intlShape} from 'react-intl'
import {shape} from 'prop-types'

import messages from '../locale/en.json'

export const testId = 'test id'

// Instantiate router context
const router = {
  history: {...new BrowserRouter().history},
  route: {
    location: {
      path: '/location',
      href: '/location',
    },
    match: {
      path: '/location',
      params: {
        id: testId,
      },
    },
  },
}

// const createContext = () => ({
//   context: {router},
//   childContextTypes: {router: shape({})},
// })

const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

const createMixedContext = ({
  router,
  context,
  childContextTypes,
  ...additionalOptions
} = {}) => ({
  context: Object.assign({}, {router}, context, {intl}),
  childContextTypes: Object.assign(
    {},
    {router: shape({})},
    {intl: intlShape},
    childContextTypes,
  ),
  ...additionalOptions,
})

export function mountWrap(node) {
  return mount(node, createMixedContext({router}))
}

// export function shallowWrap(node) {
//   return shallow(node, createContext())
// }
