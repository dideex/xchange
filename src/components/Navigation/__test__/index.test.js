import React from 'react'
import {Provider as MobxProvider} from 'mobx-react'

import Component from '../index'
import UserStore from '../../../store/User'
import {fakeUser} from '../../../helpers/fixtures'
import {mountWrap} from '../../../helpers/router-intl-context'

describe('Navigation: index', () => {
  let userStore = new UserStore()
  beforeEach(() => {
    userStore = new UserStore()
  })
  describe('Markup', () => {
    it('Basic markup', () => {
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )

      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('Component behavriour', () => {
  })
})
