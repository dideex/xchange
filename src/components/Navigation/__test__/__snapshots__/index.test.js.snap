// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation: index Markup Basic markup 1`] = `"<nav class=\\"  css-oet3qe\\"><svg style=\\"width: 215px;\\" class=\\"\\" viewBox=\\"0 0 216 62\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M82.4766 32.3146C82.4031 34.6807 81.7417 36.8249 80.566 38.5995C79.3902 40.448 77.7735 41.8529 75.6424 42.8141C73.5113 43.8493 71.0863 44.2929 68.4408 44.2929C63.9582 44.2929 60.4309 42.8141 57.8589 39.8565C55.2869 36.8989 53.9642 32.7582 53.9642 27.4345V25.7339C53.9642 22.4065 54.5521 19.4489 55.7278 16.9349C56.9036 14.421 58.5938 12.4985 60.7249 11.0937C62.9294 9.68878 65.4279 9.02332 68.2939 9.02332C72.4091 9.02332 75.7159 10.1324 78.2879 12.2767C80.7864 14.4949 82.2561 17.4525 82.5501 21.2974H74.3932C74.3197 19.2271 73.8053 17.7483 72.85 16.7871C71.8947 15.8998 70.3515 15.4561 68.3674 15.4561C66.3098 15.4561 64.84 16.1955 63.8847 17.7483C62.9294 19.301 62.415 21.7411 62.415 25.1423V27.5824C62.415 31.2055 62.8559 33.8673 63.8112 35.4201C64.6931 36.9728 66.3098 37.7862 68.5143 37.7862C70.4249 37.7862 71.8212 37.3425 72.85 36.4552C73.8788 35.5679 74.3932 34.1631 74.4667 32.1667H82.4766V32.3146Z\\" style=\\"fill: #191716;\\"></path><path d=\\"M93.72 21.2236C95.4101 19.0793 97.5412 18.0441 100.187 18.0441C102.979 18.0441 105.11 18.8575 106.506 20.5581C107.903 22.1848 108.638 24.6988 108.711 27.9521V43.9972H100.775V28.1C100.775 26.7691 100.481 25.7339 99.9662 25.1424C99.4518 24.4769 98.4965 24.1812 97.1738 24.1812C95.5571 24.1812 94.3813 24.6988 93.72 25.7339V43.9972H85.7835V7.84033H93.6465V21.2236H93.72Z\\" style=\\"fill: #191716;\\"></path><path d=\\"M127.303 43.9233C127.009 43.4058 126.788 42.6663 126.568 41.6312C125.098 43.4797 123.114 44.367 120.469 44.367C118.117 44.367 116.06 43.6276 114.369 42.2227C112.679 40.7439 111.871 38.9693 111.871 36.7511C111.871 34.0153 112.9 31.871 114.884 30.4662C116.941 29.0613 119.881 28.3219 123.702 28.3219H126.127V27.0649C126.127 24.6988 125.098 23.5897 123.114 23.5897C121.277 23.5897 120.322 24.5509 120.322 26.3994H112.459C112.459 23.9594 113.488 21.963 115.545 20.4103C117.603 18.8575 120.248 18.1181 123.482 18.1181C126.715 18.1181 129.287 18.9314 131.124 20.4842C132.961 22.0369 133.99 24.2552 133.99 26.991V38.156C133.99 40.5221 134.357 42.2966 135.092 43.5536V43.9233H127.303ZM122.306 38.7475C123.261 38.7475 124.069 38.5257 124.731 38.082C125.392 37.6384 125.833 37.1947 126.127 36.6772V32.6104H123.849C121.13 32.6104 119.734 33.8674 119.734 36.3075C119.734 37.0469 119.954 37.6384 120.469 38.082C120.983 38.5257 121.571 38.7475 122.306 38.7475Z\\" style=\\"fill: #191716;\\"></path><path d=\\"M145.674 18.4877L145.968 21.4454C147.732 19.1532 150.157 17.9702 153.17 17.9702C155.815 17.9702 157.799 18.7835 159.049 20.3362C160.371 21.889 161.033 24.329 161.106 27.5085V43.9233H153.17V27.8782C153.17 26.6212 152.876 25.66 152.361 25.0684C151.847 24.4769 150.892 24.1812 149.569 24.1812C148.026 24.1812 146.85 24.7727 146.115 26.0297V43.9972H138.252V18.4877H145.674Z\\" style=\\"fill: #191716;\\"></path><path d=\\"M164.56 31.0576C164.56 28.4697 165.001 26.1775 165.809 24.2551C166.618 22.2587 167.867 20.7799 169.41 19.6708C170.953 18.5617 172.79 18.0441 174.922 18.0441C177.567 18.0441 179.625 18.9314 181.094 20.7059L181.388 18.4877H188.516V42.962C188.516 45.1802 188.002 47.1766 186.973 48.8033C185.944 50.43 184.401 51.7609 182.491 52.6482C180.506 53.5355 178.228 53.9791 175.656 53.9791C173.819 53.9791 171.982 53.6094 170.292 52.944C168.602 52.2785 167.206 51.3173 166.324 50.2082L169.631 45.5499C171.1 47.3245 173.011 48.1378 175.436 48.1378C178.963 48.1378 180.653 46.3633 180.653 42.7402V41.9268C179.11 43.6275 177.2 44.4408 174.922 44.4408C171.835 44.4408 169.337 43.2578 167.426 40.8917C165.515 38.5256 164.56 35.3461 164.56 31.3534V31.0576ZM172.423 31.5012C172.423 33.5716 172.79 35.2722 173.599 36.4552C174.407 37.6383 175.436 38.3038 176.906 38.3038C178.596 38.3038 179.845 37.7122 180.653 36.6031V25.8818C179.919 24.6987 178.669 24.1072 176.979 24.1072C175.583 24.1072 174.481 24.7727 173.672 26.0296C172.864 27.2866 172.423 29.1351 172.423 31.5012Z\\" style=\\"fill: #191716;\\"></path><path d=\\"M205.271 44.4408C201.376 44.4408 198.216 43.2578 195.865 40.9656C193.513 38.6735 192.264 35.568 192.264 31.8709V31.2055C192.264 28.6176 192.779 26.3254 193.734 24.329C194.689 22.3326 196.085 20.7799 197.923 19.6708C199.76 18.5617 201.964 18.0441 204.463 18.0441C207.99 18.0441 210.856 19.1532 212.914 21.3714C214.971 23.5896 216 26.6951 216 30.6879V33.7195H200.348C200.642 35.1243 201.229 36.2334 202.185 37.1207C203.14 37.9341 204.389 38.3777 205.859 38.3777C208.358 38.3777 210.268 37.4904 211.738 35.7158L215.339 40.0044C214.383 41.3353 212.987 42.4444 211.15 43.2578C209.239 43.9972 207.329 44.4408 205.271 44.4408ZM204.389 24.1072C202.111 24.1072 200.715 25.6599 200.274 28.6915H208.211V28.1C208.211 26.843 207.917 25.8818 207.255 25.1424C206.594 24.4769 205.565 24.1072 204.389 24.1072Z\\" style=\\"fill: #191716;\\"></path><path d=\\"M122.085 55.7538L119.66 53.2398L119.219 52.7962L116.133 49.6907V52.7962H26.3337C21.5571 52.7962 18.3237 51.3174 17.4419 48.6555C16.6336 46.2894 17.8093 43.3318 20.2344 41.4093C23.2473 38.9693 26.5541 36.5293 28.6117 34.9765L39.5611 46.1415L43.7498 41.9269L33.4618 31.5013C35.5194 29.9486 38.5323 27.6564 41.3247 25.4382C45.7339 21.8891 47.6445 16.3435 45.9543 11.6113C45.146 9.17129 41.9861 3.47787 31.4777 3.47787H6.41907V0.372375L2.89176 3.92152L0.39325 6.43549L3.33267 9.39311L6.41907 12.4986V9.39311H31.5511C36.3277 9.39311 39.5611 10.8719 40.4429 13.5338C41.2512 15.8999 40.149 18.8575 37.6505 20.7799C34.5641 23.22 31.2572 25.734 29.2731 27.2128L16.9275 15.0126L12.7388 19.2272L24.3496 30.7619C22.2185 32.3147 19.2791 34.6068 16.4866 36.7511C12.0775 40.3002 10.1668 45.8458 11.857 50.578C12.6653 53.018 15.8252 58.7114 26.3337 58.7114H116.06V61.8169L120.248 57.5284L122.085 55.7538Z\\" style=\\"fill: #8fbe00;\\"></path></svg><nav class=\\"css-1hwc4ij\\"><a href=\\"/\\"><span>Home</span></a><a href=\\"/reservi\\"><span>Reserves</span></a><a href=\\"/o-nas\\"><span>About us</span></a><a href=\\"/faq\\"><span>FAQ</span></a><span data-testid=\\"lang-toggler\\"><span>Language</span><svg style=\\"width: 20px; margin-bottom: -2px; padding-left: 5px;\\" class=\\"\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><path style=\\"fill: #aee239;\\" d=\\"M184.9,344.1L8.2,167.4c-8.5-8.5-8.5-22.3,0-30.9l20.6-20.6c8.5-8.5,22.3-8.5,30.8,0l140.7,140.1l140.7-140.1 c8.5-8.5,22.3-8.5,30.8,0l20.6,20.6c8.5,8.5,8.5,22.3,0,30.9L215.8,344.1C207.3,352.6,193.5,352.6,184.9,344.1L184.9,344.1z\\"></path></svg></span><span class=\\"auth-menu-item\\"><svg style=\\"width: 33px;\\" class=\\"\\" viewBox=\\"0 0 33 38\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path style=\\"fill: #8fbe00;\\" d=\\"M16.5 19C21.7078 19 25.9286 14.7473 25.9286 9.5C25.9286 4.25273 21.7078 0 16.5 0C11.2922 0 7.07143 4.25273 7.07143 9.5C7.07143 14.7473 11.2922 19 16.5 19ZM23.1 21.375H21.8699C20.2346 22.132 18.4152 22.5625 16.5 22.5625C14.5848 22.5625 12.7728 22.132 11.1301 21.375H9.9C4.43437 21.375 0 25.843 0 31.35V34.4375C0 36.4043 1.58371 38 3.53571 38H29.4643C31.4163 38 33 36.4043 33 34.4375V31.35C33 25.843 28.5656 21.375 23.1 21.375Z\\"></path></svg></span></nav></nav>"`;
